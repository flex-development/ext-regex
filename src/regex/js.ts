/**
 * @file Regular EXPRESSIONS - EXT_JS_REGEX
 * @module ext-regex/regex/js
 */

/**
 * JavaScript file extension regex.
 *
 * Supported extensions:
 *
 * - `.cjs`
 * - `.js`
 * - `.jsx`
 * - `.mjs`
 *
 * Captured groups:
 *
 * - `type`: Letter between `'.'` and `'js'` in file extension
 *
 * @see https://regex101.com/r/p0ZTF7
 *
 * @const {RegExp} EXT_JS_REGEX
 */
const EXT_JS_REGEX: RegExp = /\.(?:(?<type>[cm])?js|jsx)(?=\s*$)/

export default EXT_JS_REGEX
