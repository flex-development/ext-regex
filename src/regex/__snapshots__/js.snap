// Vitest Snapshot v1

exports[`unit:EXT_JS_REGEX > .cjs > should match extension ".cjs" 1`] = `
{
  "0": ".cjs",
  "1": "c",
  "groups": {
    "type": "c",
  },
}
`;

exports[`unit:EXT_JS_REGEX > .cjs > should match module id given last extension is ".cjs" 1`] = `
{
  "0": ".cjs",
  "1": "c",
  "groups": {
    "type": "c",
  },
}
`;

exports[`unit:EXT_JS_REGEX > .cjs > should match module id given only extension is ".cjs" 1`] = `
{
  "0": ".cjs",
  "1": "c",
  "groups": {
    "type": "c",
  },
}
`;

exports[`unit:EXT_JS_REGEX > .js > should match extension ".js" 1`] = `
{
  "0": ".js",
  "1": undefined,
  "groups": {
    "type": undefined,
  },
}
`;

exports[`unit:EXT_JS_REGEX > .js > should match module id given last extension is ".js" 1`] = `
{
  "0": ".js",
  "1": undefined,
  "groups": {
    "type": undefined,
  },
}
`;

exports[`unit:EXT_JS_REGEX > .js > should match module id given only extension is ".js" 1`] = `
{
  "0": ".js",
  "1": undefined,
  "groups": {
    "type": undefined,
  },
}
`;

exports[`unit:EXT_JS_REGEX > .jsx > should match extension ".jsx" 1`] = `
{
  "0": ".jsx",
  "1": undefined,
  "groups": {
    "type": undefined,
  },
}
`;

exports[`unit:EXT_JS_REGEX > .jsx > should match module id given last extension is ".jsx" 1`] = `
{
  "0": ".jsx",
  "1": undefined,
  "groups": {
    "type": undefined,
  },
}
`;

exports[`unit:EXT_JS_REGEX > .jsx > should match module id given only extension is ".jsx" 1`] = `
{
  "0": ".jsx",
  "1": undefined,
  "groups": {
    "type": undefined,
  },
}
`;

exports[`unit:EXT_JS_REGEX > .mjs > should match extension ".mjs" 1`] = `
{
  "0": ".mjs",
  "1": "m",
  "groups": {
    "type": "m",
  },
}
`;

exports[`unit:EXT_JS_REGEX > .mjs > should match module id given last extension is ".mjs" 1`] = `
{
  "0": ".mjs",
  "1": "m",
  "groups": {
    "type": "m",
  },
}
`;

exports[`unit:EXT_JS_REGEX > .mjs > should match module id given only extension is ".mjs" 1`] = `
{
  "0": ".mjs",
  "1": "m",
  "groups": {
    "type": "m",
  },
}
`;
